### PR TITLE
Adding 'topology' as an accessible parameter for testsystems

### PR DESCRIPTION
**Pull request by jchodera**
This PR adds access to the topologies for each test system via a `testsystem.topology` property.

### CODE REVIEW

**Comment by jchodera** [comment]
OK, this seems to be working now.

Can someone else take a quick look?  @kyleabeauchamp or @jhprinz?

This adds a `.topology` property to each model system that should accurately provide an OpenMM `Topology` object. I've added a test to make sure that the `Topology` object has the same number of atoms as the OpenMM `System` object.

**Comment by kyleabeauchamp** [comment]
looking now

File: openmmtools/testsystems.py [hunk 1]
```diff
@@ -350,6 +357,19 @@ def positions(self):
         del self._positions
 
     @property
+    def topology(self):
```
Comment (kyleabeauchamp) [inline openmmtools/testsystems.py]: So with the deepcopy here, it's essentially impossible to edit the underlying topology via `testsystem.topology`.  I guess this is probably a good idea here, but it's worth thinking about.

Comment (jchodera) [inline openmmtools/testsystems.py]: Currently, `testsystem._topology` is mutable, `testsystem.topology` returns an immutable copy.

We do the same with `testsystem.system`.  

I'm happy to go either way on this.  In particular, I'd love to go with whichever behavior is the least surprising to users/developers.

**Comment by kyleabeauchamp** [comment]
I'm fine with sticking with what you've got, but IMHO we want to describe this behavior in a Notes section in a docstring somewhere.

**Comment by jchodera** [comment]
On further reflection, I think we should switch these back to being mutable, since that would likely cause less confusion.  But let's do that in a separate PR.

**Comment by jchodera** [comment]
Everything else OK?  I realize there is some inconsistency in how these topology objects are constructed in each system, but we can probably clean that in the future.

**Comment by kyleabeauchamp** [comment]
AFAIK this seems reasonable for now.

**Comment by jchodera** [comment]
I just updated the `TestSystem` docstring to be clearer, but we can change this in the PR to address #41.

**Pull request merged**